packages/messaging/ios/RNFBMessaging/RNFBMessagingModule.m

```diff
@@ -219,6 +219,19 @@ - (NSDictionary *)constantsToExport {
   return resolve(@([RCTConvert BOOL:@(notifCenter.isHeadless)]));
 }
 
+RCT_EXPORT_METHOD(completeNotificationProcessing) {
+  RNFBMessagingAppDelegate *appDelegate = [RNFBMessagingAppDelegate sharedInstance];
+  if (appDelegate.completionHandler) {
```

Comment:
I think this should be on the same queue as the initial set of completionHandler / backgroundTaskId otherwise there's a tiny chance via race condition that multiple threads are working on their state at the same time

It's this above, would that work here?

>                     dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^{`